NorthstarDLL/core/structs.h

```diff
@@ -25,38 +25,24 @@ OFFSET_STRUCT(Name)
 
 #define OFFSET_STRUCT(name) union name
 #define STRUCT_SIZE(size) char __size[size];
-#define STRUCT_FIELD_OFFSET(offset, signature)                                                                                             \
-	struct                                                                                                                                 \
-	{                                                                                                                                      \
-		char CONCAT2(pad, __LINE__)[offset];                                                                                               \
-		signature;                                                                                                                         \
-	};
+#define STRUCT_FIELD_OFFSET(offset, signature) \
+	char CONCAT2(pad, __LINE__)[offset]; \
+	signature;
 
 // Special case for a 0-offset field
 #define STRUCT_FIELD_NOOFFSET(offset, signature) signature;
 
-// Based on: https://stackoverflow.com/questions/11632219/c-preprocessor-macro-specialisation-based-on-an-argument
-// Yes, this is hacky, but it works quite well actually
-// This basically makes sure that when the offset is 0x0, no padding field gets generated
-#define OFFSET_0x0 ()
+#define FX(f,x) f x
 
-#define IIF(c) CONCAT2(IIF_, c)
-#define IIF_0(t, ...) __VA_ARGS__
-#define IIF_1(t, ...) t
+#define TEST_0 ,
 
-#define PROBE(x) x, 1
-#define MSVC_VA_ARGS_WORKAROUND(define, args) define args
-#define CHECK(...) MSVC_VA_ARGS_WORKAROUND(CHECK_N, (__VA_ARGS__, 0))
-#define DO_PROBE(offset) PROBE_PROXY(OFFSET_##offset) // concatenate prefix with offset
-#define PROBE_PROXY(...) PROBE_PRIMITIVE(__VA_ARGS__) // expand arguments
-#define PROBE_PRIMITIVE(x) PROBE_COMBINE_##x // merge
-#define PROBE_COMBINE_(...) PROBE(~) // if merge successful, expand to probe
+#define ZERO_P_I(a,b,c,...) a##c
+#define IF_ZERO(m) FX(ZERO_P_I, (NIF_, CONCAT2(TEST_, m), 1))
 
-#define CHECK_N(x, n, ...) n
+#define NIF_(t,...) t
+#define NIF_1(t,...) __VA_ARGS__
```

Comment:
The core logic of this boils down to the same thing as the original, except fewer passes are required and it does not rely on MSVC specific behavior.